### PR TITLE
Removed seconds_per_iter option, allowing the user instead of directl…

### PR DESCRIPTION
**Pull request by OhGodAPet**
…y control the number of kernel executions before hashrate is shown and new work is fetched. Added basic control over intensity - the amount of work items run by the GPU in one kernel enqueue - via the command line. Added control over both the OpenCL platform ID and device ID within that platform that is selected to mine on via the command line, as well. Finally, there's a little bit of code cleaning done, code added to fetch all platforms/devices available, and so on.

### CODE REVIEW

File: gpu-miner.c [hunk 1]
```diff
@@ -129,51 +140,155 @@ double grindNonces(size_t items_per_iter, int cycles_per_iter) {
...
 			break;
-		case 'c':
-			sscanf(optarg, "%d", &cycles_per_iter);
-			if (cycles_per_iter < 1 || cycles_per_iter > 1000) {
-				printf("Cycles per iter must be at least 1 and no more than 1000\n");
-				exit(1);
+		case 'I':
+			Intensity = strtoul(optarg, NULL, 10);		// Returns zero on error
+			
+			if(Intensity && (Intensity < MIN_INTENSITY) && (Intensity > MAX_INTENSITY))
```
Comment (DavidVorick) [inline gpu-miner.c]: Maybe I'm reading it wrong, but I think you wanted:

``` c
if (!Intensity || Intensity < MIN_INTENSITY || Intensity > MAX_INTENSITY) {
```

Comment (OhGodAPet) [inline gpu-miner.c]: Hm... you're right! Thanks.

File: gpu-miner.c [hunk 2]
```diff
@@ -100,9 +109,11 @@ double grindNonces(size_t items_per_iter, int cycles_per_iter) {
 		if (ret != CL_SUCCESS) { printf("failed to write to targmobj buffer: %d\n", ret); exit(1); }
 
 		// Execute OpenCL kernel as data parallel
+		// Note that minimum intensity is 8, making the local
+		// worksize always divisible by the global worksize
```
Comment (lukechampine) [inline gpu-miner.c:113]: Do you mean the global worksize is divisible by the local worksize?
But in the lshift above you use `Intensity - 1` -- doesn't that make `GlobalSize = 128` when the Intensity is 8?

File: gpu-miner.c [hunk 3]
```diff
@@ -129,51 +140,155 @@ double grindNonces(size_t items_per_iter, int cycles_per_iter) {
...
 	port_number = "9980";
-	while ( (c = getopt(argc, argv, "hc:s:p:")) != -1) {
+	while ( (c = getopt(argc, argv, "hI:p:d:C:P:")) != -1) {
 		switch (c) {
 		case 'h':
 			printf("\nUsage:\n\n");
-			printf("\t c - cycles per iter: Number of workloads hashing gets split into each iteration\n");
-			printf("\t\tIncrease this if your computer is freezing or locking up\n");
+			printf("\t I - intensity: This is the amount of work sent to the GPU in one batch.\n");
+			printf("\t\tInterpretation is 2^intensity; the default is 17. Lower if GPU crashes or\n");
```
Comment (lukechampine) [inline gpu-miner.c]: default is 16, not 17

File: gpu-miner.c [hunk 4]
```diff
@@ -129,51 +140,155 @@ double grindNonces(size_t items_per_iter, int cycles_per_iter) {
...
+		exit(1);
+	}
+	
+	// Done. Return the platform ID and device ID object desired, free lists, and return.
+	*OCLPlatform = AllPlatforms[PlatformIdx];
+	*OCLDevice = AllDevices[DeviceIdx];
+}
+	
+int main(int argc, char *argv[])
+{
```
Comment (lukechampine) [inline gpu-miner.c]: please conform to our current style (`{` on the same line)

File: gpu-miner.c (hunk 2 above)
Comment (OhGodAPet) [inline gpu-miner.c:113]: Yes; it was really late that night - I did an off-by-one in my head. You're correct; 1 << 7 == 0x80 or 128.

File: gpu-miner.c (hunk 4 above)
Comment (OhGodAPet) [inline gpu-miner.c]: Noted for the future.

**Comment by DavidVorick** [comment]
PR got merged from a different branch.

Thanks for this. You're on our contributors list now: https://github.com/NebulousLabs/Sia-GPU-Miner/graphs/contributors

**Pull request closed without merging**